pandas/_libs/tslibs/nattype.pyx

```diff
@@ -1133,6 +1133,22 @@ cdef inline bint checknull_with_nat(object val):
     """
     return val is None or util.is_nan(val) or val is c_NaT
 
+cdef inline bint is_dt64nat(object val):
+    """
+    Utility to check if val is np.datetime64("NaT").
+    """
+    if util.is_datetime64_object(val):
+        return get_datetime64_value(val) == NPY_NAT
+    return False
+
+cdef inline bint is_td64nat(object val):
+    """
+    Utility to check if val is np.timedelta64("NaT").
```

Comment:
can you clarify that "is" in this context doens't mean what it does in python code